index.html

```diff
@@ -0,0 +1,464 @@
+              If an existing web app client is open it is brought to focus and
+              navigated to the launch's target URL. If there are no existing web
+              app clients the [=route to/new-client=] behaviour is used instead.
+            </dd>
+            <dt>
+              <dfn data-dfn-for="route to">existing-client-retain</dfn>
+            </dt>
+            <dd>
+              If an existing web app client is open it is brought to focus but
+              not navigated to the launch's target URL, instead the target URL
+              is communicated via {{LaunchParams}} . If there are no existing
+              web app clients the [=route to/new-client=] behaviour is used
+              instead.
+            </dd>
+          </dl>
+        </p>
+        <p>
+          To <dfn>process the `route_to` member</dfn>, given [=ordered
+          map=] |json_launch_handler:ordered map|, [=ordered map=]
+          |manifest_launch_handler:ordered map|, run the following:
+        </p>
+        <ol class="algorithm">
+          <li>If |json_launch_handler|["route_to"] does not [=map/exist=],
+          return.
+          </li>
+          <li>If the type of |json_launch_handler|["route_to"] is
+          [=list=]:
+            <ol class="algorithm">
+              <li>[=list/For each=] |entry| of |json_launch_handler|["route_to"]:
+                <ol class="algorithm">
+                  <li>If the type of |entry| is not [=string=], continue.
+                  </li>
+                  <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |entry| and return.
+                  </li>
+                </ol>
+              </li>
+            </ol>
+          </li>
+          <li>If the type of |json_launch_handler|["route_to"] is not [=string=], return.
+          </li>
+          <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |json_launch_handler|["route_to"].
+          </li>
+        </ol>
+        <p class="note">
+          `route_to` accepts a list of strings as a future proofing mechanism. If a new [=route to target=] is added in the future but has not yet been implemented by all user agents, web developers can cater to the old user agents by providing fallback [=route to target=] values in a list format. User agents will use the first value that is valid to them.
+        </p>
+        <p class="note">
+          For example: if "some-new-behaviour" is added in the future a site developer can specify "[=route to/existing-client-navigate=]" to be the fallback with the following:
+        </p>
+        <pre class="note json">
+          {
+            "launch_handler": {
+              "route_to": ["some-new-behaviour", "existing-client-navigate"]
+            }
+          }
+        </pre>
+      </section>
+    </section>
+    <section>
+      <h2>
+        Handling Web App Launches
```

Comment:
This section is fundamental enough that it should be in the manifest spec (eventually). It can start here but we should try to work it upwards (possibly via Manifest Incubations) because it isn't just an "extension" of the manifest, it very much captures what it means to link into an app.

Manifest spec already has something to say on this topic: [§6.2 Deep links](https://www.w3.org/TR/appmanifest/#deep-links):

> An application context can be instantiated through a deep link, in which case, the manifest is applied and the deep link is loaded within the context of a web application.

I've never been satisfied with that tiny section. I would eventually want to replace that section with a top-level section called Handling Web App Launches which is in a position to link together all the concepts of file handling, share target, link capturing, etc, and tie them into launch handling.

Also note that [§1.17.4 Applying the manifest](https://www.w3.org/TR/appmanifest/#applying) has some normative text around launching an app window and navigating to it, which we'll want to subsume with all this text:

> If an [application context](https://www.w3.org/TR/appmanifest/#dfn-application-context) is created as a result of the user agent being asked to [navigate](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigate) to a [deep link](https://www.w3.org/TR/appmanifest/#dfn-deep-link), the user agent MUST immediately [navigate](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigate) to the [deep link](https://www.w3.org/TR/appmanifest/#dfn-deep-link) with historyHandling set to "replace". Otherwise, when the [application context](https://www.w3.org/TR/appmanifest/#dfn-application-context) is created, the user agent MUST immediately [navigate](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigate) to the [start URL](https://www.w3.org/TR/appmanifest/#dfn-start-url) with historyHandling set to "replace".

For now, no action required.